pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -0,0 +1,389 @@
+        }
+        setCommands = {
+            "setup": "TRSE %s,SR,{ch},HT,%s,HV,%s",
+            "level": "{ch}:TRLV %s",
+            "coupling": "{ch}:TRCP %s",
+            "slope": "{ch}:TRSL %s",
+            "mode": "TRMD %s",
+        }
+        setValues = {  # For a given change in conf dict,find the relevant command to be called
+            "setup": ["source", "type", "hold_type", "hold_value1"],
+            "level": ["level"],
+            "coupling": ["coupling"],
+            "slope": ["slope"],
+            "mode": ["mode"],
+        }
+        if kwargs.get("source") is not None:
+            self.id = kwargs["source"]
+        changedValues = [key for key in kwargs if trigger_config_dict[key] != kwargs[key]]
+        processToChange = [
+            key for key in setValues if any([value in changedValues for value in setValues[key]])
+        ]
+        for changedKey in changedValues:
+            trigger_config_dict[changedKey] = kwargs[changedKey]
+        for processKey in processToChange:
+            self.write(
+                setCommands[processKey] % setProcesses[processKey](trigger_config_dict),
+            )
+        self.trigger_config_dict = self.get_trigger_config()
+        statusFlag = self.trigger_config_dict == trigger_config_dict
+        return statusFlag
+
+
+class SDS1072CML(SCPIMixin, Instrument):
+    """
+    ==============================================
+    Represents the SIGLENT SDS1072CML Oscilloscope
+    ==============================================
+    """
+
+    def __init__(self, adapter, name="Siglent SDS1072CML Oscilloscope", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    channel_1 = Instrument.ChannelCreator(VoltageChannel, "1")
+    channel_2 = Instrument.ChannelCreator(VoltageChannel, "2")
+    trigger = Instrument.ChannelCreator(TriggerChannel, "")
+
+    time_division = Instrument.control(
+        ":TDIV?",
+        ":TDIV %.2eS",
+        "Control the time division to the closest possible value,rounding downwards.",
+        validator=truncated_range,
+        values=[5e-9, 50],
+        get_process=lambda v: float(v.split(" ", 1)[-1][:-1]),
+    )
+
+    status = Instrument.measurement(
+        "SAST?",
+        "Get the sampling status of the scope (Stop, Ready, Trig'd, Armed)",
+        get_process=lambda v: v.split(" ", 1)[-1],
+    )
+
+    internal_state = Instrument.measurement(
+        "INR?",
+        "Get the scope's Internal state change register and clears it.",
+        get_process=lambda v: v.split(" ", 1)[-1],
+    )
+
+    is_ready = Instrument.measurement(
+        "SAST?",
+        "Get whether the scope is ready for the next acquisition (bool)",
+        get_process=lambda v: v.split(" ", 1)[-1] in ["Stop", "Ready", "Armed"],
+    )
+
+    def wait(self, time):
+        """Stop the scope from doing anything until it has completed the current acquisition (p.146)
+        param time: time in seconds to wait for
+        """
```

Comment:
Docstrings should have (according to PEP 257) a single sentence in the first line (terminated with a period). Then an empty line, afterwards a summary and/or the parameters.
Also, for sphinx (our documentation generator), you have to start a parameter line with a colon:
```suggestion
        """Stop the scope from doing anything until it has completed the current acquisition.

        :param time: time in seconds to wait for
        """
```

why do you reference the manual (page 146)? is there any relevant information you better shared here?